src/navigationStore.js

```diff
@@ -888,11 +888,22 @@ class NavigationStore {
   };
 
   dispatch = (action) => {
-    if (this.externalDispatch) {
-      this.externalAction = action;
-      this.externalDispatch(action);
-    } else if (this._navigator) {
-      this._navigator.dispatch(action);
+    if (action.params && action.params.forceReplace) {
+      if (this.externalDispatch) {
+        this.externalAction = action;
+        this.externalDispatch(action);
+      } else if (this._navigator) {
+        this._navigator.dispatch(action);
+      }
+      return;
+    }
+    if (action.routeName !== this.currentScene) {
+      if (this.externalDispatch) {
+        this.externalAction = action;
+        this.externalDispatch(action);
+      } else if (this._navigator) {
+        this._navigator.dispatch(action);
+      }
```

Comment:
What if I have a scene which accepts params, and can navigate to itself, but with a different param? For example, imagine app where you read blog posts. The scene where you read a certain blog post is called "Post". Inside of it, you see a hyperlink which basically takes you to another blog post, so the route name doesn't change. If this change gets approved, all those links stop working.

You need to add a logic that checks how many milliseconds are passed since user last navigated. If it's really too fast, (less than... 500ms? I'm a fast guy, I tap fast, so maybe 300ms?) - then cancel navigating.

But I wouldn't bother. What was the problem you had?